### PR TITLE
Allow console name to be translated

### LINKED ISSUE #2756: "Console" title in locale file
### Feature request

**Feature description**
Put "Console" string to locale file. This is that "Console" that appears to player when he receive PM from console.

**How the feature is useful**
This is useful when staff want to say something to the player on behalf of the console. For example, notify the player of some important event but also want to customize Console name to more friendly and interesting name.

### PR DESCRIPTION
**Pull request by Haarolean**
Fixes #2756. 

Notes:
- I've kept original name for internal purposes (storing bans, etc), just in case user wants to rename it to something unbearable;
- User can message console via both names (since he can try both, duh);
- Some refactoring. Just a little.

### CODE REVIEW

File: Essentials/src/com/earth2me/essentials/Console.java [hunk 1]
```diff
@@ -2,15 +2,17 @@
 
 import com.earth2me.essentials.messaging.IMessageRecipient;
 import com.earth2me.essentials.messaging.SimpleMessageRecipient;
-
```
Comment (Octol1ttle) [inline Essentials/src/com/earth2me/essentials/Console.java]: Unnecessary diff

File: Essentials/src/com/earth2me/essentials/Console.java [hunk 2]
```diff
@@ -2,15 +2,17 @@
...
 
+import static com.earth2me.essentials.I18n.tl;
+
 
 public final class Console implements IMessageRecipient {
     public static final String NAME = "Console";
+    public static final String DISPLAY_NAME = tl("consoleName");
     private static Console instance; // Set in essentials
-    
+
```
Comment (Octol1ttle) [inline Essentials/src/com/earth2me/essentials/Console.java]: Unnecessary diff

File: Essentials/src/com/earth2me/essentials/Console.java [hunk 3]
```diff
@@ -39,39 +41,47 @@ public CommandSender getCommandSender() {
...
-    @Override public void sendMessage(String message) {
+    @Override
+    public void sendMessage(String message) {
         getCommandSender().sendMessage(message);
     }
 
-    @Override public boolean isReachable() {
+    @Override
+    public boolean isReachable() {
         return true;
```
Comment (Octol1ttle) [inline Essentials/src/com/earth2me/essentials/Console.java]: Why did you change that?

Comment (Haarolean) [reply Essentials/src/com/earth2me/essentials/Console.java]: Auto formatting. I've checked the projects, turns out mostly there's a line break after annotations. Since there's no clear code style I decided to keep it this way. If you mind, I can revert.

File: Essentials/src/com/earth2me/essentials/Console.java (hunk 2 above)
Comment (Haarolean) [reply Essentials/src/com/earth2me/essentials/Console.java]: Auto formatting. There was 4 black spaces on this empty line.

File: Essentials/src/messages.properties [hunk 4]
```diff
@@ -2,144 +2,144 @@
 #version: ${full.version}
 # Single quotes have to be doubled: ''
 # by:
-action=\u00a75* {0} \u00a75{1}
-addedToAccount=\u00a7a{0} has been added to your account.
-addedToOthersAccount=\u00a7a{0} added to {1}\u00a7a account. New balance\: {2}
+action=\u00A75* {0} \u00A75{1}
+addedToAccount=\u00A7a{0} has been added to your account.
+addedToOthersAccount=\u00A7a{0} added to {1}\u00A7a account. New balance\: {2}
```
Comment (Octol1ttle) [inline Essentials/src/messages.properties]: This whole file is full of changes like this. I don't see any reason behind this

File: Essentials/src/com/earth2me/essentials/Console.java (hunk 1 above)
Comment (Haarolean) [reply Essentials/src/com/earth2me/essentials/Console.java]: That's auto formatting again. Usually there's just an empty line between usual imports and static ones. With all due respect, do you have any notes on the actual code?

File: Essentials/src/messages.properties (hunk 4 above)
Comment (Haarolean) [reply Essentials/src/messages.properties]: Oh my, that's my IDE settings with native-to-ascii conversion. Fixed!

File: Essentials/src/com/earth2me/essentials/Console.java (hunk 1 above)
Comment (Octol1ttle) [reply Essentials/src/com/earth2me/essentials/Console.java]: Actual code seems good. (to me atleast)

**Review by Octol1ttle — CHANGES_REQUESTED** [review]
I recommend you turn off auto-formatting while PR-ing something. We all have different codestyles, but it's best to keep the original one

File: Essentials/src/com/earth2me/essentials/Console.java (hunk 3 above)
Comment (pop4959) [reply Essentials/src/com/earth2me/essentials/Console.java]: Please turn off auto formatting and keep a minimal diff when working on Essentials PRs. Reverting it would be ideal. 😄

Comment (Haarolean) [reply Essentials/src/com/earth2me/essentials/Console.java]: Waiting on #3701 to rebase.

Comment (pop4959) [reply Essentials/src/com/earth2me/essentials/Console.java]: no rush, PRs usually take a little while before merging anyways

happy oktoberfest still 😄

**Comment by Haarolean** [comment]
@md678685 would you remove the rebase label please? I've rebased already.

File: EssentialsXMPP/src/com/earth2me/essentials/xmpp/Commandxmpp.java [hunk 5]
```diff
@@ -25,7 +25,9 @@ protected void run(final Server server, final CommandSource sender, final String
         }
 
         final String message = getFinalArg(args, 1);
-        final String senderName = sender.isPlayer() ? ess.getUser(sender.getPlayer()).getDisplayName() : Console.NAME;
+        final String senderName = sender.isPlayer()
+                ? ess.getUser(sender.getPlayer()).getDisplayName()
+                : Console.DISPLAY_NAME;
```
Comment (JRoy) [inline EssentialsXMPP/src/com/earth2me/essentials/xmpp/Commandxmpp.java]: please keep this to one line

File: Essentials/src/com/earth2me/essentials/commands/Commandmsg.java [hunk 6]
```diff
@@ -40,8 +40,9 @@ public void run(final Server server, final CommandSource sender, final String co
         }
 
         // Sending messages to console
-        if (args[0].equalsIgnoreCase(Console.NAME)) {
-            final IMessageRecipient messageSender = sender.isPlayer() ? ess.getUser(sender.getPlayer()) : Console.getInstance();
+        if (args[0].equalsIgnoreCase(Console.NAME) || args[0].equalsIgnoreCase(Console.DISPLAY_NAME)) {
+            final IMessageRecipient messageSender = sender.isPlayer() ? ess.getUser(sender.getPlayer()) :
+                    Console.getInstance();
```
Comment (JRoy) [inline Essentials/src/com/earth2me/essentials/commands/Commandmsg.java]: please keep this to one line

File: Essentials/src/messages.properties [hunk 7]
```diff
@@ -973,3 +973,4 @@ years=years
 youAreHealed=\u00a76You have been healed.
 youHaveNewMail=\u00a76You have\u00a7c {0} \u00a76messages\! Type \u00a7c/mail read\u00a76 to view your mail.
 xmppNotConfigured=XMPP is not configured properly. If you do not know what XMPP is, you may wish to remove the EssentialsXXMPP plugin from your server.
+consoleName=Console
```
Comment (JRoy) [inline Essentials/src/messages.properties]: Please sort this alphabetically

File: Essentials/src/com/earth2me/essentials/commands/Commandbanip.java [hunk 8]
```diff
@@ -50,18 +51,19 @@ public void run(final Server server, final CommandSource sender, final String co
...
 
         for (final Player player : ess.getServer().getOnlinePlayers()) {
             if (player.getAddress().getAddress().getHostAddress().equalsIgnoreCase(ipAddress)) {
                 player.kickPlayer(banDisplay);
             }
         }
 
-        ess.broadcastMessage("essentials.banip.notify", tl("playerBanIpAddress", senderName, ipAddress, banReason));
+        ess.broadcastMessage("essentials.banip.notify", tl("playerBanIpAddress",
+                senderDisplayName, ipAddress, banReason));
```
Comment (JRoy) [inline Essentials/src/com/earth2me/essentials/commands/Commandbanip.java]: please keep this to one line

File: Essentials/src/com/earth2me/essentials/commands/Commandban.java [hunk 9]
```diff
@@ -50,16 +51,17 @@ public void run(final Server server, final CommandSource sender, final String co
...
-        server.getLogger().log(Level.INFO, tl("playerBanned", senderName, user.getName(), banDisplay));
+        server.getLogger().log(Level.INFO, tl("playerBanned", senderDisplayName, user.getName(), banDisplay));
 
         if (nomatch) {
             sender.sendMessage(tl("userUnknown", user.getName()));
         }
 
-        ess.broadcastMessage("essentials.ban.notify", tl("playerBanned", senderName, user.getName(), banReason));
+        ess.broadcastMessage("essentials.ban.notify", tl("playerBanned",
+                senderDisplayName, user.getName(), banReason));
```
Comment (JRoy) [inline Essentials/src/com/earth2me/essentials/commands/Commandban.java]: please keep this to one line

**Review by JRoy — CHANGES_REQUESTED** [review]

File: EssentialsXMPP/src/com/earth2me/essentials/xmpp/Commandxmpp.java (hunk 5 above)
Comment (Haarolean) [reply EssentialsXMPP/src/com/earth2me/essentials/xmpp/Commandxmpp.java]: I don't see any line lengths/wrapping guidelines in #3701, so I used common sense here -- wrapped everything which was above 120 symbols. A successful build indicates that no checkstyle rule was violated. I don't mind applying required length if there's any project-scope preference, not a personal opinion.
@md678685, any thoughts on this?

File: Essentials/src/messages.properties (hunk 7 above)
Comment (Haarolean) [reply Essentials/src/messages.properties]: Sorry, didn't notice. Done.

File: EssentialsXMPP/src/com/earth2me/essentials/xmpp/Commandxmpp.java (hunk 5 above)
Comment (JRoy) [reply EssentialsXMPP/src/com/earth2me/essentials/xmpp/Commandxmpp.java]: Pretty sure the Essentials code doesn't split lines unless creating a builder class, I could be wrong however.

Comment (pop4959) [reply EssentialsXMPP/src/com/earth2me/essentials/xmpp/Commandxmpp.java]: Yeah just as a convention, I don't think Essentials normally wraps lines. Just because no checkstyle rule was violated (it was also added extremely recently) doesn't mean we can't still discuss it. 😄 Maybe let MD comment on this since he's managing those rules but I think one line is more correct here as far as I know.

Comment (Haarolean) [reply EssentialsXMPP/src/com/earth2me/essentials/xmpp/Commandxmpp.java]: @mdcfe bump?

Comment (mdcfe) [reply EssentialsXMPP/src/com/earth2me/essentials/xmpp/Commandxmpp.java]: One line is preferable where possible - many of these line breaks feel arbitrary.

File: Essentials/src/com/earth2me/essentials/EssentialsUpgrade.java [hunk 10]
```diff
@@ -680,11 +680,8 @@ public void banFormatChange() {
     }
 
     private void updateBan(final String playerName, final String banReason, final Long banTimeout) {
-        if (banTimeout == 0) {
-            Bukkit.getBanList(BanList.Type.NAME).addBan(playerName, banReason, null, Console.NAME);
-        } else {
-            Bukkit.getBanList(BanList.Type.NAME).addBan(playerName, banReason, new Date(banTimeout), Console.NAME);
-        }
+        final Date date = banTimeout == 0 ? null : new Date(banTimeout);
+        Bukkit.getBanList(BanList.Type.NAME).addBan(playerName, banReason, date, Console.NAME);
```
Comment (JRoy) [inline Essentials/src/com/earth2me/essentials/EssentialsUpgrade.java]:
```suggestion
        Bukkit.getBanList(BanList.Type.NAME).addBan(playerName, banReason, banTimeout == 0 ? null : new Date(banTimeout), Console.NAME);
```

**Review by JRoy — APPROVED** [review]

**Review by JRoy — CHANGES_REQUESTED** [review]
whoops see above requested changes

File: Essentials/src/com/earth2me/essentials/EssentialsUpgrade.java (hunk 10 above)
Comment (Haarolean) [reply Essentials/src/com/earth2me/essentials/EssentialsUpgrade.java]: @JRoy what's wrong with inline variables?

Comment (JRoy) [reply Essentials/src/com/earth2me/essentials/EssentialsUpgrade.java]: You don't need to declare another variable here when you're just going to declare and use it once

Comment (Haarolean) [reply Essentials/src/com/earth2me/essentials/EssentialsUpgrade.java]: Ah, nvm, I see.
I've extracted it coz the line was too long and it's easier to read. But whatevs.

**Review by JRoy — APPROVED** [review]
good for merge after gradle pr

**Review by mdcfe — APPROVED** [review]
